File: frontend/src/types/dealStage.ts
export interface DealStage {
  id: string
  name: string
  description?: string
  order: number
  color?: string
  isDefault: boolean
  isActive: boolean
  createdAt: string
  updatedAt: string
}

export interface CreateDealStageDto {
  name: string
  description?: string
  order: number
  color?: string
  isDefault?: boolean
}

export interface UpdateDealStageDto {
  name?: string
  description?: string
  order?: number
  color?: string
  isDefault?: boolean
  isActive?: boolean
}

File: backend/CrmApp.Domain/Entities/DealStage.cs
namespace CrmApp.Domain.Entities;

public class DealStage : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public int Order { get; set; }
    public string? Description { get; set; }
    public string? Color { get; set; }
    public bool IsDefault { get; set; }
    
    // Navigation properties
    public ICollection<Opportunity> Opportunities { get; set; } = new List<Opportunity>();
}
